### ISSUE TITLE
Loader `options` values of type `Function` are not present on loader options

### ISSUE THREAD

**Issue by briandipalma** [mzgoddard/hard-source-webpack-plugin#149]
I have a simple loader, this is the code

```javascript
const loaderUtils = require("loader-utils");

module.exports = function patchLoader(moduleSource) {
  let options = loaderUtils.getOptions(this);

  return options.appendModulePatch(this, moduleSource);
};
```

Which appends a patch to a module. The loader configuration uses a function

```javascript
     {
        test: /\.js$/,
        loader: "@caplin/patch-loader",
        options: {
          appendModulePatch: appendModulePatch()
        }
      }
```

The second time a build is run, after the cache has been created we are getting an error due to `appendModulePatch` being `undefined`. That's because the Webpack config wasn't fully copied over to the loader config so `loader-utils` fails.

I can work around this issue by adding this patch to the loader.

```javascript
  if (options.appendModulePatch === undefined) {
    options = this.options.module.rules.find(({ loader }) => loader === "@caplin/patch-loader").options;
  }
```

I was wondering if I'm missing something or is this a bug?

**Comment by mzgoddard** [comment]
@briandipalma I tried to reproduce this the other day and couldn't produce a failure from what you are describing. Can you make a small repository that produces the error? That'd help me fix this.

---

Though really I think this is likely a bug with some other plugin or webpack feature you are using. `hard-source` doesn't manipulate how webpack builds modules. It caches the built output and uses a complete stand-in for the previously built modules. It does a bunch of checks and if they fail has webpack rebuild the module. `hard-source` shouldn't be involved with the loaders and their options reaching the loader.

My first thought was if you are using happy-pack or thread-loader that would cause the issue as they run your loader in a separate process and don't have a way to transport the function. As well though, your work around wouldn't be functional in that case either, so I guess you aren't using those tools.

**Comment by mzgoddard** [comment]
Nevermind I just reproduced it.

**Comment by mzgoddard** [comment]
Ah. It has to do with the cached resolver.

**Comment by mzgoddard** [comment]
Fixed in #153
